File: 03_parcleable/app/src/main/java/com/bomnie/a03_parcleable/MainActivity.java
package com.bomnie.a03_parcleable;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), MenuActivity.class);

                ArrayList<String> names = new ArrayList<String>();
                names.add("Son");
                names.add("Harry ");

                intent.putExtra("name", names);
                // 인텐트를 전달하면서 키 값과 객체이름을 같이 보냄

                SimpleData data = new SimpleData(77, "Hello");
                intent.putExtra("data", data);
                // 기본적으로 전달하는 데이터에는 객체를 넣을 수 없지만
                // parcelable 로 구현한 SimpleData 객체는 넣을 수 있음!
                startActivityForResult(intent,  101 );
            }
        });
    }
}

File: KotlinRecyclerViewApp/app/src/main/java/com/bomnie/kotlinrecyclerviewapp/MainActivity.kt
package com.bomnie.kotlinrecyclerviewapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 멤버 변수를 만들 필요 없이
        // id가 btn 인 버튼에 클릭리스너 추가하기
        // [코틀린은 id가 참조변수!]
        btn.setOnClickListener(object : View.OnClickListener{
            override fun onClick(v: View?) {
                //SecondActivity 를 실행해주는 Intent 객체 생성
                var intent= Intent(this@MainActivity, SecondActivity::class.java)
                startActivity(intent)
            }
        })


    }

    fun clickExit(view: View) {
        finish() // 액티비티 종료
    }
}
